circuit-retained/src/androidInstrumentedTest/kotlin/com/slack/circuit/retained/android/RetainedStateRestorationTester.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.circuit.retained.android
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.test.junit4.ComposeContentTestRule
+import com.slack.circuit.retained.LocalRetainedStateRegistry
+import com.slack.circuit.retained.RetainedStateRegistry
+import com.slack.circuit.retained.RetainedValueProvider
+import com.slack.circuit.retained.rememberRetained
+
+/**
+ * Helps to test the retained state restoration for your Composable component.
+ *
+ * Instead of calling [ComposeContentTestRule.setContent] you need to use [setContent] on this
+ * object, then change your state so there is some change to be restored, then execute
+ * [emulateRetainedInstanceStateRestore] and assert your state is restored properly.
+ *
+ * Note that this tests only the restoration of the local state of the composable you passed to
+ * [setContent] and useful for testing [rememberRetained] integration. It is not testing the
+ * integration with any other life cycles or Activity callbacks.
+ */
+// TODO recreate for more realism? Need to save the content function to do that, or call it after
+// TODO make this available in test utils?
+class RetainedStateRestorationTester(private val composeTestRule: ComposeContentTestRule) {
```

Comment:
nit: not for this PR, but do you think we should make this available as a test utility somewhere in circuit-test?

Reply:
Moving it to circuit-test could make it more accessible for others, similar to `StateRestorationTester`, which seems like a good idea. But I’m not quite sure how frequently `emulateRetainedInstanceStateRestore` would be used.

Reply:
yeah we can save that for another time!